### PR TITLE
feat: added raw query calls

### PR DESCRIPTION
**Pull request by JoseARomeroJ**
Added raw query support: https://github.com/uptrace/bun/issues/593

### CODE REVIEW

**Comment by vmihailenco** [comment]
Hey,

The suggested API looks like this:

```go
err := db.Raw(query, args...).Scan(ctx, &strct)
err := db.Raw(query, args...).Scan(ctx, &slice)
err := db.Raw(query, args...).Scan(ctx, &var1, &var2)
```

The implementation should look like this:

```go
func (db *DB) Raw(query string, args ...interface{}) *RawQuery {
    return &RawQuery{
        db: db,
        query: query,
        args: args,
    }
}

type RawQuery struct {
    db *DB
    query string
    args ...interface{}
}

func (q *RawQuery) Scan(ctx context.Context, dest ...interface{}) error {
    // extract and reuse Scan implementation from SelectQuery.Scan
    // to format the query, use query := db.format(q.query, q.args)
    return scanModel(...)
}
```

Does that make sense?

**Comment by JoseARomeroJ** [comment]
Did you mean something like that?

File: query_raw.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	query := q.db.format(q.query, q.args)
+	_, err = q.scan(ctx, q, query, model, true)
+	return err
+}
+
+func (q *RawQuery) AppendQuery(fmter schema.Formatter, b []byte) ([]byte, error) {
+	query := []byte(fmter.FormatQuery(q.query, q.args))
+	b = append(b, query...)
+
+	return b, nil
```
Comment (vmihailenco) [inline query_raw.go:45]: This can be `return fmter.AppendQuery(b, q.query, q.args), nil`

**Comment by vmihailenco** [comment]
Very good - thanks :+1:

**Pull request merged**